### PR TITLE
Remove Match dependency in circshift perf

### PR DESCRIPTION
**Pull request by hayd**
cc @jrevels

### CODE REVIEW

**Comment by jrevels** [comment]
Ah, I didn't dig deep enough to see how trivial the dependency on Match.jl was. Good call, I'll port this version over to BaseBenchmarks.jl.

File: test/perf/threads/lbm3d/circshift.jl [hunk 1]
```diff
@@ -11,26 +10,38 @@ function circshift3d1!(a,index,vec)
...
-        @match vec[i] begin
-            1  => shift1!(a,nx,ny,nz,index,i)
-           -1  => shiftm1!(a,nx,ny,nz,index,i)
+        vi = vec[i]
+        if vi == 1
+            shift1!(a,nx,ny,nz,index,i)
+        elseif vi == -1
+            shiftm1!(a,nx,ny,nz,index,i)
+        else
+            throw(ArgumentError)
```
Comment (jrevels) [inline test/perf/threads/lbm3d/circshift.jl]: You actually don't want to throw errors in the non-matching case; the old behavior is just to do nothing if `vi` isn't either `1` or `-1`.

Comment (hayd) [inline test/perf/threads/lbm3d/circshift.jl]: Should it ever be called with something other than 1 or -1. It seems to me that ArgumentError is more correct. :/

Comment (jrevels) [inline test/perf/threads/lbm3d/circshift.jl]: The comment that describes this function is misleading - `vec` can contain `0` values (e.g. no shift). See [here](https://github.com/JuliaLang/julia/blob/hayd-patch-1/test/perf/threads/lbm3d/lbm3d.jl#L27).

**Comment by kshyatt** [comment]
Can we close this?

**Pull request closed without merging**